core/src/main/java/org/kiwix/kiwixmobile/core/main/AnimatedProgressBar.kt

```diff
@@ -0,0 +1,238 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.animation.ObjectAnimator
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Rect
+import android.os.Bundle
+import android.os.Parcelable
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.animation.Animation
+import android.view.animation.DecelerateInterpolator
+import android.view.animation.Transformation
+import android.widget.LinearLayout
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.styleable
+
+class AnimatedProgressBar : LinearLayout {
```

Comment:
Look up the annotation `JvmOverloads`, you will be able to specify only one constructor.

Special care needs to be taken when using this with views as some are written poorly and do slightly different things depending on which constructor is invoked so always check the class you are extending, at least for the correct default arguments